packages/grid/x-data-grid-pro/src/components/GridScrollArea.tsx

```diff
@@ -124,24 +124,17 @@ function GridScrollAreaRaw(props: ScrollAreaProps) {
 
       offset = (offset - CLIFF) * SLOP + CLIFF;
 
-      clearTimeout(timeout.current);
       // Avoid freeze and inertia.
-      timeout.current = setTimeout(() => {
+      timeout.start(0, () => {
         apiRef.current.scroll({
           left: scrollPosition.current.left + offset,
           top: scrollPosition.current.top,
         });
       });
     },
-    [scrollDirection, apiRef],
+    [scrollDirection, apiRef, timeout],
```

Comment:
One thing I don't like is that eslint doesn't recognize `useTimeout` as a React ref, so it requires this.

There is a react issue for this but it's been open forever: https://github.com/facebook/react/issues/16873

Reply:
I think it's fine, especially with the ESlint auto-fix turned on.